index.js

```diff
@@ -103,15 +103,38 @@ module.exports = (input, opts) => {
 	]);
 
 	if (runPublish) {
-		tasks.add({
-			title: 'Publishing package',
-			skip: () => {
-				if (pkg.private) {
-					return 'Private package: not publishing to npm.';
+		tasks.add([
+			{
+				title: 'Publishing package using Yarn',
+				enabled: () => opts.yarn === true,
+				skip: () => {
+					if (pkg.private) {
+						return 'Private package: not publishing to Yarn.';
+					}
+				},
+				task: () => {
+					const args = ['publish', '--new-version', input];
+
+					if (opts.tag) {
+						args.push('--tag', opts.tag);
+					}
+
+					return exec('yarn', args).catch(err => {
+						throw new Error(err);
+					});
```

Comment:
Just noticed this. Why do you catch the error, rewrap it in an error, and throw it again?